packages/wallet/src/redux/fundingState/indirectFunding/state.ts

```diff
@@ -0,0 +1,102 @@
+import { SharedIndirectFundingState, INDIRECT_FUNDING } from '../shared/state';
+export { SharedIndirectFundingState, INDIRECT_FUNDING };
+
+// ChannelFundingStatus
+export const FUNDER_CHANNEL_DOES_NOT_EXIST = 'FUNDER_CHANNEL_DOES_NOT_EXIST';
+export const FUNDER_CHANNEL_EXISTS = 'FUNDER_CHANNEL_EXISTS';
+export const CHANNEL_FUNDED = 'CHANNEL_FUNDED';
+
+// Funding status
+export type ChannelFundingStatus =
+  | typeof FUNDER_CHANNEL_DOES_NOT_EXIST
+  | typeof FUNDER_CHANNEL_EXISTS
+  | typeof CHANNEL_FUNDED;
+
+export interface BaseIndirectFundingState extends SharedIndirectFundingState {
+  channelFundingStatus: ChannelFundingStatus;
+}
+
+export interface FunderChannelDoesNotExist extends BaseIndirectFundingState {
+  channelFundingStatus: typeof FUNDER_CHANNEL_DOES_NOT_EXIST;
+}
+
+export interface FunderChannelExists extends BaseIndirectFundingState {
```

Comment:
This might be better as `ChannelNotFunded` (by funder channel), since the funder channel also exists when you're in the `ChannelFunded` state.